apps/torus-worker/src/common/index.ts

```diff
@@ -246,20 +342,51 @@ export async function getVotesOnPending(
 }
 
 export async function getCadreVotes(): Promise<VoteByKey[]> {
-  const votes = await queryTotalVotesPerCadre();
+  const [error, votes] = await tryAsyncLoggingRaw(queryTotalVotesPerCadre());
+
+  if (error) {
+    log(
+      `Error querying total votes per cadre: ${error instanceof Error ? error.message : JSON.stringify(error)}`,
+    );
+    return [];
+  }
+  if (!votes) {
+    log("No votes found");
+    return [];
+  }
+
   return votes;
 }
 
 export async function getCadreThreshold() {
```

Comment:
An example of the broken typing is provided here. This function should return `Promise<number>`, not `Promise<number | undefined>`. It is expected to work. For it to fail, something very unexpected and beyond our control, such as a database issue, must occur. In this case, we can log the error, but we need to propagate it further instead of dismissing it like this (the `vote_threshold` should have the proper type inferred, when the TODOs I've added are fixed):

```javascript
// please don't do this
if (!vote_threshold) {
    log(`No vote threshold found`);
    return;
}
```

If there's an expected error, let's represent it in the return type. The function should return the same kind of [err, val] tuple instead of `number | undefined`.

cc @EdSDR